site/app/models/Gradeable.php

```diff
@@ -1339,4 +1339,181 @@ public function calculateLateStatus($checkTeam = true, $user_id = null) {
         }
     }
 
+    public function getGradingSectionsForUser(User $user) {
+        if ($this->getPeerGrading() && $user->getGroup() === 4) {
+            $users = $this->core->getQueries()->getPeerAssignment($this->getId(), $user->getId());
+            //TODO: Peer grading team assignments
+            return new GradingSection($this->core, $this, "Peer", [$user], $users, []);
+        } else {
+            $users = [];
+            $teams = [];
+
+            if ($this->isGradeByRegistration()) {
+                $section_names = $user->getGradingRegistrationSections();
+
+                if ($this->isTeamAssignment()) {
+                    foreach ($section_names as $section) {
+                        $teams[$section] = [];
+                    }
+                    $all_teams = $this->core->getQueries()->getTeamsByGradeableAndRegistrationSections($this->getId(), $section_names);
+                    foreach ($all_teams as $team) {
+                        /** @var Team $team */
+                        $teams[$team->getRegistrationSection()][] = $team;
```

Comment:
Could avoid this loop if you juts made the return value of `getTeamsByGradeableAndRegistrationSections` return the array indexed like this?